go/interpreter/lfvm/interpreter_test.go

```diff
@@ -562,34 +562,37 @@ func TestStepsProperlyHandlesJUMP_TO(t *testing.T) {
 	}
 }
 
-func TestStepsDetectsNonExecutableCode(t *testing.T) {
+func TestSteps_DetectsNonExecutableCode(t *testing.T) {
+
 	nonExecutableOpCodes := []OpCode{
 		INVALID,
 		NOOP,
 		DATA,
 	}
-
-	re := regexp.MustCompile(`^op\(0x[0-9a-fA-F]{2}\)`)
-	for op := OpCode(0); op < numOpCodes; op++ {
-		if re.MatchString(op.String()) {
-			nonExecutableOpCodes = append(nonExecutableOpCodes, op)
+	undefinedOpCodeRegex := regexp.MustCompile(`^op\(0x[0-9a-fA-F]+\)`)
+	isUndefined :=
+		func(op OpCode) bool {
+			return undefinedOpCodeRegex.MatchString(op.String())
 		}
-	}
+	nonExecutableOpCodes = append(nonExecutableOpCodes, allOpCodesWhere(isUndefined)...)
 
 	for _, opCode := range nonExecutableOpCodes {
-		ctxt := getEmptyContext()
-		ctxt.params = tosca.Parameters{
-			Input:  []byte{},
-			Static: false,
-			Gas:    10,
-			Code:   []byte{0x0},
-		}
-		ctxt.code = []Instruction{{opCode, 0}}
+		t.Run(opCode.String(), func(t *testing.T) {
+			ctxt := getEmptyContext()
+			ctxt.params = tosca.Parameters{
+				Input:  []byte{},
+				Static: false,
+				Code:   []byte{0x0},
+			}
+			ctxt.code = []Instruction{{opCode, 0}}
+			ctxt.stack.stackPointer = 50
+			ctxt.gas = 1 << 32
```

Comment:
`params`, `stack` size and `gas` are actually irrelevant for this test
```suggestion
			ctxt := getEmptyContext()
			ctxt.code = []Instruction{{opCode, 0}}
```